__tests__/app.test.js

```diff
@@ -149,4 +148,88 @@ describe("GET /api", () => {
+                    .send({ comment })
+                    .expect(201)
+                    .then(({body}) => {
+                        expect(body.comment.author).toBe(comment.username);
+                        expect(body.comment.body).toBe(comment.body);
+                        expect(body.comment.article_id).toBe(3);
+                        expect(body.comment.votes).toBe(0);
+                        expect(body.comment).toHaveProperty("comment_id", expect.any(Number));
+                        expect(body.comment).toHaveProperty("created_at", expect.any(String));
+                    })
+                })
+                test("404: Should return \"Article ID not found\" if article ID doesn't exist.", () => {
+                    const comment = {
+                        username: "butter_bridge",
+                        body: "I really miss chocolate Paddlepops, and they only cost 50c when I was a kid."
+                    }
+                    return request(app)
+                    .post('/api/articles/999/comments')
+                    .set('Content-Type', 'application/json')
+                    .send({ comment })
+                    .expect(404)
+                    .then(({body}) => {
+                        expect(body.msg).toBe("Article ID not found");
+                    })
+                })
+                test("400: Should return \"Bad request\" using existing error handles if given invalid id.", () => {
+                    const comment = {
+                        username: "butter_bridge",
+                        body: "I really miss chocolate Paddlepops, and they only cost 50c when I was a kid."
+                    }
+                    return request(app)
+                    .post('/api/articles/invalid_id/comments')
+                    .set('Content-Type', 'application/json')
+                    .send({ comment })
+                    .expect(400)
+                    .then(({body}) => {
+                        expect(body.msg).toBe("Bad request");
+                    })
+                })
+                test("400: Should return \"Bad request\" using existing error handles if comment data types are incorrect.", () => {
+                    const comment = {
+                        username: "butter_bridge",
+                        body: 12345
+                    }
+                    return request(app)
+                    .post('/api/articles/invalid_id/comments')
+                    .set('Content-Type', 'application/json')
+                    .send({ comment })
+                    .expect(400)
+                    .then(({body}) => {
+                        expect(body.msg).toBe("Bad request");
+                    })
+                })
+                test("401: Should return \"Unauthorised username\" if username doesn't exist in database.", () => {
+                    const comment = {
+                        username: "freezypop",
```

Comment:
this will be a 404

Reply:
I've refactored this into the other foreign key contraints error handles.